todo_app/data/trello_items.py

```diff
@@ -0,0 +1,198 @@
+    
+    url = f"https://api.trello.com/1/lists/{list_id}/cards"
+
+    response = requests.request(
+        "GET",
+        url,
+        headers=headers,
+        params=query)
+
+    items = []
+    
+    if response.status_code == 200:
+        cards = json.loads(response.text)
+
+        for c in cards:
+            i = Item(c['idShort'], c['name'], list_name)
+            items.append(i)        
+    else:
+        print(f"_get_cards_in_list request has failed. Status code: {response.status_code}")
+    
+    return items
+
+def _get_card_id(list_id, id_short : int):
+    
+    url = f"https://api.trello.com/1/lists/{list_id}/cards"
+
+    response = requests.request(
+        "GET",
+        url,
+        headers=headers,
+        params=query)
+
+    if response.status_code == 200:
+        cards = json.loads(response.text)
+       
+        for c in cards:
+            if c['idShort'] == id_short:
+                return c['id']
+    else:
+        print(f"_get_card_id request has failed. Status code: {response.status_code}")
+    
+    return ""
+
+def _add_card(list_id, card_name):
+
+    url = "https://api.trello.com/1/cards"
+
+    query['name'] = card_name
+    query['idList'] = list_id
+
+    response = requests.request(
+        "POST",
+        url,
+        headers=headers,
+        params=query
+    )
+
+    print(response.request.url)
+    print(response.request.body)
+    print(response.request.headers)
+
+def get_items():
+    """
+    Fetches items from the Trello API.
+
+    Returns:
+        list: The list of saved items.
+    """
+    all_items = []
+    lists_names = _get_lists()
+
+    for n in lists_names:
+        cards = _get_cards_in_list(_get_list_id(n),n)
+        for c in cards:
+            all_items.append(c)
+     
+    return  all_items
+
+def add_item(title):
+    """
+    Adds a new item with the specified title to the Trello board.
+
+    Args:
+        title: The title of the item.
+
+    Returns:
+        item: The saved item.
+    """
+    todo_list_id = _get_list_id("To Do")
+    _add_card(todo_list_id, title)
+
+def move_card_to_new_list(card_short_id, current_list_name, new_list_name):
+    """
+    Moves an specific card from one list to another
+
+    Args:
+        card_short_id: short id of the card to move e.g. 14
+        current_list_name: name of the list that the card to move belongs to 'To Do'
+        new_list_name: name of the list that the card to move needs be moving to 'Done'
+
+    Returns:
+        item: void
+    """
+    current_list_id = _get_list_id(current_list_name)
+    new_list_id = _get_list_id(new_list_name)    
+    card_id = _get_card_id(current_list_id, card_short_id)
+
+    #TODO: check errors
+
+    url = f"https://api.trello.com/1/cards/{card_id}"
+
+    # add destination list
+    query['idList'] = new_list_id
+    
+    print(f"moving card with Short ID {card_short_id} and long id {card_id} from old list id {current_list_id} to new list id {new_list_id}")
```

Comment:
While this feels like useful logging - is it worth the extra network call/code to get the current list id? I don't think there's a right answer, but an interesting question to consider!

Reply:
it was a debugging message so I'll remove it but I don't think it implies any extra network call, all the parameters printed are already known at that point and they are necessary to resolve the query... did I understand your comment correctly?